types/foundry/client/apps/app.d.ts

```diff
@@ -393,8 +393,8 @@ declare global {
     }
 
     interface ApplicationPosition {
```

Comment:
This was written before ApplicationV2 and was meant to represent the position of Application, there was no type/interface definition and whoever wrote the foundry types for PF2e ended up declaring their own. ApplicationV2 came along and did define a ApplicationPosition but it's not compatible with what Application uses, the main thing being that Application does not support using "auto" for it's width.

It would probably be better to remove the ApplicationPosition and properly declare it in the correct place so you aren't using ApplicationV2's position data for Application by mistake.

https://github.com/7H3LaughingMan/foundry-pf2e/commit/a9759b6695c71d71bce26906836966957df0e8e4#diff-8a61b31dd83b09ab3b6d7b2a0adaa14ded534c8e1fb6b54a54f783553d0e1280

Reply:
I would have to check to be clear, and it might be part of a v13 research, but I did confirm that changing string to "auto" breaks any use of "auto" in the actual sheets. Typescript can't tell that its "auto" fsr:

![image](https://github.com/user-attachments/assets/a2357797-f746-4c03-86ff-2cb660bbaf27)

Reply:
I am not sure where PartySheetV2 is located but that looks like an issue with it's DEFAULT_OPTIONS.

The Foundry types are outdated mostly because the only things that are needed are updated. They also declare types/interfaces that aren't declared to try and "simplify" things, and over the months/years they have worked their way through a bunch of stuff and trying to correct it is turning into a mess.

When the Foundry types originally declared ApplicationPosition there was no actual ApplicationPosition as ApplicationV2 wasn't a thing, with ApplicationV2 it ended up declaring an ApplicationPosition for itself and now the Foundry types are trying to use it for both Application and ApplicationV2 but they aren't compatible with each other since they expect different things.

I am working on re-writing the Foundry types to get them up to date for module development since some modules use other parts of Foundry which aren't used by PF2e, this has caused some problems here and there due to how ingrained some of these types have become. But with the v13 work being done I can't push these fixes since I can fix the problem with v12 as they pop-up they will break things with v13 so I am going to wait and see about submitting one big PR once there is a v13 system release.

Reply:
That might be wise. I don't wanna fix that distinction in this PR, but thanks for letting me know about it.

This was a PR to try to make the tab interface of appv2 compatible with svelte, and is more of a preview at this time. I'm coming up with alternative ideas as I continue, though I'm a bit blocked because I noticed that with our current implementation we may have to resort to prop drilling.

I experimented with this PR and putting foundryApp in the svelte context, and that actually seems to work! However now I have to prop drill `isEditable`, which is a DocumentSheet property, and having the mixin be aware of whether its a document sheet or not is above the intended scope.

So I'm experimenting with alternatives, and while I have a simpler way to handle tabGroups in my head now, it doesn't really solve prop drilling.